snapcraft/storeapi/errors.py

```diff
@@ -369,6 +370,22 @@ def __init__(self, result):
         super().__init__()
 
 
+class StoreServerError(StoreError):
+
+    fmt = '{what}: {error_text} (code {error_code}).\n{action}'
+
+    def __init__(self, response):
+        what = 'The store encountered a server error'
+        error_text = responses[response.status_code].lower()
+        error_code = response.status_code
```

Comment:
if you switched these lines around, you could use `error_code` to lookup `responses`